File: recyclerview-swipe/src/main/java/android/support/v7/widget/helper/CompatItemTouchHelper.java
/*
 * Copyright 2016 Yan Zhenjie
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package android.support.v7.widget.helper;

/**
 * Created by Yan Zhenjie on 2016/8/1.
 */
public class CompatItemTouchHelper extends ItemTouchHelper {

    public CompatItemTouchHelper(ItemTouchHelper.Callback callback) {
        super(callback);
    }

    /**
     * Developer callback which controls the behavior of ItemTouchHelper.
     *
     * @return {@link Callback}
     */
    public ItemTouchHelper.Callback getCallback() {
        return mCallback;
    }
}

File: recyclerview-swipe/src/main/java/android/support/v7/widget/helper/CompatItemTouchHelper.kt
/*
 * Copyright 2016 Yan Zhenjie
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package android.support.v7.widget.helper

/**
 * Compat item touch helper.
 *
 * Created by Yan Zhenjie on 2016/8/1.
 */
open class CompatItemTouchHelper(callback: ItemTouchHelper.Callback) : ItemTouchHelper(callback) {

  /**
   * Developer callback which controls the behavior of ItemTouchHelper.
   */
  val callback: ItemTouchHelper.Callback
    get() = mCallback

}
